src/runtime_src/ert/scheduler/CMakeLists.txt

```diff
@@ -5,15 +5,31 @@
 
 if (${ERT_BUILD_ALL} STREQUAL "yes")
 
-function(GetBSP BSP)
+function(CopyBSP BSP)
+ message(STATUS "Copying ${BSP} from $ENV{ERT_BSP_DIR}")
+ file(COPY $ENV{ERT_BSP_DIR}/${BSP} DESTINATION ${CMAKE_CURRENT_BINARY_DIR})
+endfunction()
+
+function(DownloadBSP BSP)
  set(URL https://github.com/Xilinx/ERT-BSP/raw/main/BSPs/${BSP})
+ message(STATUS "Downloading ${BSP} from ${URL}")
  FILE(DOWNLOAD ${URL} ${CMAKE_CURRENT_BINARY_DIR}/${BSP} STATUS dlstatus)
  if(NOT dlstatus EQUAL 0)
   message(FATAL_ERROR "ERROR: Failed to retrieve files from: ${URL}\n"
                        "   error code: ${dlstatus}\n")
  endif()
 endfunction()
 
+function(GetBSP BSP)
+ if(EXISTS ${CMAKE_CURRENT_BINARY_DIR}/${BSP})
+  message(STATUS "${BSP} exists, skip fetching")
```

Comment:
Is there any chance the current already downloaded BSP could be old and cause debug nightmare?

Reply:
@stsoe

These files are related to legacy Alveo shells. They are tested, fixed and we do not expect any one to make any change to them.

In case we, somehow, ever need to change one of them, we could encode version of these binary files in their file names to avoid any confusion. I.e., in case there is ever need a change, we could:
1. Push a new version of BSP binary file to BSP repo with a new file name, like xxxx_v2.
2. Modify the XRT make file to use new file where necessary.
3. Keep old files name the same so that older XRT can still compile with them and sees no change.

Sounds OK?